src/unimodal.jl

```diff
@@ -0,0 +1,81 @@
+"""
+```
+t = find_threshold(Unimodal(),  histogram, edges)
+```
+
+This method generates a threshold using Rosin's algorithm for unimodal images.
+It assumes that:
+
+*  The histogram is unimodal
+*  There is always at least one bin that has a frequency of 0, to be used as
+the minimum. If not, the algorithm will use the last bin.
+*  If there are multiple zero bins, the algorithm will use the first zero bin.
+
+# Arguments
+
+The function arguments are described in more detail below.
+
+##  `histogram`
+
+An `AbstractArray` storing the frequency distribution.
+
+##  `edges`
+
+An `AbstractRange` specifying how the intervals for the frequency distribution
+are divided.
+
+## Reference
+
+Rosin, P. (2001). Unimodal thresholding. Pattern Recognition, [online] 34(11), pp.2083-2096. Available at: https://users.cs.cf.ac.uk/Paul.Rosin/resources/papers/unimodal2.pdf.
+
+"""
+
+
+#= Calculates the distance between the line calculated with slope m,
+y-intercept c, and a point wit \h co-ordinates x₀, y₀
+=#
+function calc_dist(m::Real, c::Real, x₀::Real, y₀::Real)
+    return abs(m*x₀ + -1*y₀ + c) / sqrt((m)^2 + (-1)^2)
+end
+
+
+function find_threshold(algorithm::UniThreshold,  histogram::AbstractArray, edges::AbstractRange)
+
+    # Find the index & value of bin with highest value
+    max_val = findmax(histogram)[1]
```

Comment:
Instead of 
```julia
    max_val = findmax(histogram)[1]
    max_index = findmax(histogram)[2]
```

you can do

```julia
max_val, max_index = findmax(histogram)
```